### PR TITLE
MINOR: INFO log4j when request re-join

### PR DESCRIPTION
**Pull request by guozhangwang**
While debugging a rebalance scenario I found that inside rejoinNeededOrPending when we trigger rebalance due to metadata or subscription changes it is not logged, and hence it's actually a bit tricky to find out the reason of the triggered rebalance. I'm adding two INFO log4j entries to fill in the gap.

Other requestRejoin() calls are already covered.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
@vvcephei @abbccdda

**Comment by guozhangwang** [comment]
test this

**Comment by guozhangwang** [comment]
test this

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java [hunk 1]
```diff
@@ -770,12 +770,18 @@ public boolean rejoinNeededOrPending() {
...
+                    assignmentSnapshot, metadataSnapshot);
+
             requestRejoin();
             return true;
         }
 
         // we need to join if our subscription has changed since the last join
         if (joinedSubscription != null && !joinedSubscription.equals(subscriptions.subscription())) {
+            log.info("Requesting to re-join the group and trigger rebalance since the subscription has changed from {} to {}",
+                    joinedSubscription, subscriptions.subscription());
```
Comment (abbccdda) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java]: nit: fix alignment.

**Review by abbccdda — COMMENTED** [review]
Rest LGTM

**Pull request merged**